sr/comp/scores.py

```diff
@@ -457,3 +492,36 @@ def __init__(
         """
         The match with the highest id for which we have score data.
         """
+
+    def get_scores(self, match: Match) -> Optional[MatchScore]:
+        """
+        Get the scores for a given match.
+
+        Parameters
+        ----------
+        match : sr.comp.match_period.Match
+            A match.
+
+        Returns
+        -------
+        MatchScore | None
+            An object describing the scores for the match, if scores have been
+            recorded yet. Otherwise None.
```

Comment:
My sense is that calling this function only really on matches which do have recorded scores is strongly the normal case, so I wonder whether raising a `LookupError` of some description might be a better fit than making the return value optional?

Reply:
The immediate use will be within SRComp HTTP's `match_json_info`, to replace the implementation that that has.
In that scenario (and I suspect the majority case) the caller doesn't know ahead of time whether scores have been recorded for the match (and won't really know until they ask). The goal of this API is to avoid the caller needing to look directly at the lower level points mappings on the `BaseScores` objects for each of the stages of the competition, so I wouldn't want to end up encouraging callers to go to those just to get a list of scored matches.

I wonder if an alternative interface here would help -- perhaps if the `Scores` class supported `Mapping` protocol members?

Reply:
On discussing this in a recent doing, I'm going to stick with this API. Our reasoning was that since you have a match object the match must actually exist, it's just that the scores aren't available yet and they will be at some point.